source.go

```diff
@@ -712,6 +720,64 @@ func fillDefaults(s *Source) {
 		fillDefaults(&s.Build.Source)
 	case s.Inline != nil:
 	}
+
+}
+
+func (git *SourceGit) fillDefaults(generators []*SourceGenerator) {
+	if git == nil {
+		return
+	}
+
+	u, err := url.Parse(git.URL)
+	if err != nil {
+		// This should be validated elsewhere
+		return
+	}
+
+	for _, generator := range generators {
+		generator.fillDefaults(u.Host, &git.Auth)
```

Comment:
Is this so we don't have to duplicate auth in source and generator config?
Can you add a comment on that?